draft-ietf-quic-tls.md

```diff
@@ -1281,6 +1280,52 @@ Never assume that because it isn't in the security considerations section it
 doesn't affect security.  Most of this document does.
 
 
+## Replay Attacks with 0-RTT
+
+As described in Section 8 of {{!TLS13}}, use of TLS early data comes with an
+exposure to replay attack.  The use of 0-RTT in QUIC is similarly vulnerable to
+replay attack.
+
+Endpoints MUST implement and use the replay protections described in {{!TLS13}},
+however it is recognized that these protections are imperfect.  Therefore,
+additional consideration of the risk of replay are needed.
+
+QUIC is not inherently vulnerable to replay attack.  The management of QUIC
+protocol state based on the frame types defined in {{QUIC-TRANSPORT}} is not
+vulnerable to replay.  Processing of QUIC frames is idempotent and cannot result
+in invalid connection states if frames are reordered or lost.  QUIC connections
+do not produce effects that last beyond the lifetime of the connection, except
+for those produced by the application protocol that QUIC serves.
```

Comment:
```suggestion
for those produced by TLS or the application protocol that QUIC serves.
```

(addresses @mikkelfj's point below)

Reply:
I realize that I had typed a response to this, but didn't post it...

This change would be a mistake.  TLS, like QUIC itself, is not the application protocol.  While it is true that the effects of tickets and address validation tokens last beyond the connection (and this might have some implications for the text), they are not effects that an application needs to worry about.

I would prefer to add "TLS session tickets and address validation tokens might be carried between connections, but these mechanisms are distinct from application state and require independent protection against reuse."  Or something to that effect.